File: src/main/java/com/cuupa/classificator/services/kb/MetaData.java
package com.cuupa.classificator.services.kb;

import java.util.ArrayList;
import java.util.List;

public class MetaData {

	private String name;

	private List<Token> token = new ArrayList<>();
	
	public void setName(String name) {
		this.name = name;
	}

	public void addToken(Token token) {
		this.token.add(token);
	}

	public void extract(String text) {
		for (Token token2 : token) {
			List<String> tokenValue = token2.tokenValue;
			
			for (int i = 0; i < tokenValue.size(); i++) {
				String string = tokenValue.get(i).split("\\[")[0];
			}
			
			token2.match(text);
		}
	}

	private Token compileToken(Token token2) {
		
		return null;
	}
}

File: src/main/java/com/cuupa/classificator/services/kb/Metadata.kt
package com.cuupa.classificator.services.kb

import com.cuupa.classificator.services.kb.semantic.token.Token

data class Metadata(val name: String, val value: String) : SemanticResultData() {
    override fun addToken(token: Token) {
        //noting to do here
    }
}